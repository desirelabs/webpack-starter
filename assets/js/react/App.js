// @flow
import React from 'react'

type Props = {
  name: string
}

const App = ({name}: Props) => (
  <div style={style}>
    <h1>Hello {name}!</h1>
  </div>
)

const style = {
  display: 'flex',
  flexFlow: 'column nowrap',
  alignItems: 'center'
}

export default App
