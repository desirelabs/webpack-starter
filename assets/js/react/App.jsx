import React from 'react'

class App extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      name: '',
      people: {}
    }

    this.setData = this.setData.bind(this)
  }

  componentWillMount () {
    this.setData()
  }

  setData () {
    this.setState({
      name: 'React'
    })
  }

  render () {
    return (
      <h1>Hello {this.state.name}!</h1>
    )
  }
}

export default App
