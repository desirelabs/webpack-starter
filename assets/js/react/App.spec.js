
import React from 'react'
import App from './App'
import renderer from 'react-test-renderer'

describe('App', () => {
  /**
   * Snapshot testing
   */
  test('Should render', () => {
    const wrapper = renderer.create(<App />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
