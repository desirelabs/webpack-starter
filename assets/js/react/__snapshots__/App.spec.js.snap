// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexFlow": "column nowrap",
    }
  }
>
  <h1>
    Hello 
    !
  </h1>
</div>
`;
